File: lumodule/lineup/lineup.py
class EasyVar(Exec):
    def __init__(self, typer):
        super(EasyVar, self).__init__()
        self.type = typer
        self.value = None
        self.command = {
            "set": self.set,
            "get": self.get,
            "descript": self.descript
        }

    def set(self, p1):
        self.value = self.type(p1)

    def get(self):
        return self.value

    def descript(self, value):
        return self.type(value)


class StringL(EasyVar):
    def __init__(self, *args):
        super(String, self).__init__(str)
        self.value = " ".join(args)
        self.command = {
            "set": self.set,
            "get": self.get,
            "count": self.count,
            "linecut": self.linecut,
            "fresh": self.fresh,
            "descript": self.descript
        }

    def count(self, p1=None):
        if p1 is None:
            return len(self.string)
        else:
            return len(p1)

    def linecut(self, p1=None):
        return module.linup.List(*p1.split("\n"))

    def fresh(self, string, number):
        return module.lineup.List(*[string[i:i+int(number)] for i in range(0, len(string), int(number))])


class IntL(EasyVar):
    def __init__(self, args=0):
        super(Int, self).__init__(int)
        self.value = int(args)


class FloatL(EasyVar):
    def __init__(self, args=0):
        super(Float, self).__init__(float)
        self.value = float(args)


class Operation(Exec):
    def __init__(self):
        super(Operation, self).__init__()
        self.command = {
            "+": self.plus,
            "-": self.moins,
            "*": self.multi,
            "/": self.divide,
            "//": self.dividei,
            "**": self.exp,
            "%": self.modulo
        }

    def plus(self, n1, n2):
        return n1 + n2

    def moins(self, n1, n2):
        return n1 - n2

    def multi(self, n1, n2):
        return n1 * n2

    def divide(self, n1, n2):
        return n1 / n2

    def dividei(self, n1, n2):
        return n1 // n2

    def exp(self, n1, n2):
        return n1 ** n2

    def modulo(self, n1, n2):
        return n1 % n2


class Condition(Exec):
    def __init__(self):
        super(Condition, self).__init__()
        self.command = {
            "=": self.equal,
            ">=": self.much_equal,
            "<=": self.low_equal,
            "!=": self.not_equal
        }

    def equal(self, n1, n2):
        return n1 == n2

    def much_equal(self, n1, n2):
        return n1 >= n2

    def low_equal(self, n1, n2):
        return n1 <= n2

    def not_equal(self, n1, n2):
        return n1 != n2


class Process(Exec):
    def __init__(self, global_variable, global_class):
        super(Process, self).__init__()
        self.global_variable = global_variable
        self.global_class = global_class
        self.command = {
            "if": self.if_pro,
            "while": self.while_pro,
            "multiple": self.multiple
        }

    def if_pro(self, *args):
        # {} {} {}
        if args[0].execute(self.global_variable, self.global_class):
            return args[1].execute(self.global_variable, self.global_class)
        else:
            if len(args) == 3:
                return args[2].execute(self.global_variable, self.global_class)
        return None

    def multiple(self, *args):
        if isinstance(args[0], str):
            data = []
            for i in args[1].decoded_line:
                data.append(i[1].execute(self.global_variable, self.global_class))
            return data[int(args[0])]
        return [i[1].execute(self.global_variable, self.global_class) for i in args[0].decoded_line]

    def while_pro(self, *arg):
        print("no")


class ListL(Exec):
    def __init__(self, *args):
        super(List, self).__init__()
        self.list = [i for i in args]
        self.command = {
            "get": self.get,
            "set": self.set,
            "setnew": self.setnew,
            "display": self.display
        }

    def get(self, index, lister=None):
        if lister is None:
            return self.list[int(index)]
        else:
            return lister.get(index)

    def set(self, new):
        self.list.append(new)

    def setnew(self, new):
        if isinstance(new, list):
            self.list = new
        else:
            self.list = new.list

    def display(self):
        return "[" + ", ".join([str(i) for i in self.list]) + "]"

File: lumodule/lineup/lineup.php
<?php

class EasyVar extends Execution {
    public $type;
    public $value;


    public function __construct($typer) {
        parent::__construct();
        $this->type = $typer;
        $this->value = null;
    }
	
	public function set($parameters) {
		return $this->value;
	}
	
	public function get($parameters) {
		$this->value = $this->type($parameters[0]);
	}
	
	public function descript($parameters) {
		$data = $this->type;
		return $data($parameters[0]);
	}
}

class IntL extends EasyVar {
    public function __construct() {
        parent::__construct("intval");
    }
}

class FloatL extends EasyVar {	
    public function __construct() {
        parent::__construct("floatval");
    }
}

class StringL extends EasyVar {
	public function __construct() {
		parent::__construct("strval");
	}
	
	public function count($parameters) {
		if (count($parameters) >= 1) {
			return count($parameters[0]);
		} else {
			return count($this->value);
		}
	}
	
	public function linecut($parameters) {
		global $module;
		return $module->lineup->ListL(explode(PHP_EOF, $parameters[0]));
	}
	
	public function fresh($parameters) {
		global $module;
		return new $module->lineup->ListL(str_split($parameters[0], $parameters[1]));
	}
}

class Operation extends Execution {
	public function __construct() {
		parent::__construct();
	}
	
	public function plus($parameters) {
		return $parameters[0] + $parameters[1];
	}
	
	public function minus($parameters) {
		return $parameters[0] - $parameters[1];
	}
	
	public function multi($parameters) {
		return $parameters[0] * $parameters[1];
	}
	
	public function divide($parameters) {
		return $parameters[0] / $parameters[1];
	}
	
	public function divideFloat($parameters) {
		return floatval($parameters[0]) / $parameters[1];
	}
	
	public function exp($parameters) {
		return $parameters[0] ** $parameters[1];
	}
	
	public function modulo($parameters) {
		return $parameters[0] % $parameters[1];
	}
}

class Condition extends Execution {
	public function __construct() {
		parent::__construct();
	}
	
	public function equal($parameters) {
		return ($parameters[0] == $parameters[1]);
	}
	
	public function Iequal($parameters) {
		return ($parameters[0] >= $parameters[1]);
	}
	
	public function equalI($parameters) {
		return ($parameters[0] <= $parameters[1]);
	}
	
	public function Nequal($parameters) {
		return ($parameters[0] != $parameters[1]);
	}
}

class ListL extends Execution {	
	public $listing;
	
	public function __construct() {
		parent::__construct();
		$this->listing = array();
	}
	
	public function get($parameters) {
		if (isset($parameters[1])) {
			return $parameters[1]->get(array($parameters[0]));
		} else {
			return $this->listing[$parameters[0]];
		}
	}
	
	public function set($parameters) {
		$this->listing[] = $parameters[0];
	}
	
	public function setnew($parameters) {
		return;
	}
	/*
	def setnew(self, new):
        if isinstance(new, list):
            self.list = new
        else:
            self.list = new.list
			*/
	
	public function display($parameters) {
        return "[".implode(", ", $this->listing)."]";
		
	} 
}


class Process extends Execution {
	public $global_variable;
	public $global_class;
	
	public function __construct(&$global_variable, &$global_class) {
		$this->global_variable = &$global_variable;
		$this->global_class = &$global_class;
	}
	
	public function if($parameters) {
		if ($parameters[0]->execute($this->global_variable, $this->global_class)) {
			return $parameters[1]->execute($this->global_variable, $this->global_class);
		} else {
			if (count($parameters) == 3) {
				return $parameters[2]->execute($this->global_variable, $this->global_class);
			}
		}
		return null;
	}
	
	public function multiple($parameters) {
		$data = array();
		if (is_string($parameters[0])) {
			$t = null;
			foreach ($parameters[1]->decoded_line as $value) {
				$data[] = $value[1]->execute($this->global_variable, $this->global_class);
			}
			return $data[$t];
		}
		foreach ($parameters[0]->decoded_line as $value) {
			$data[] = $value[1]->execute($this->global_variable, $this->global_class);
		}
		return $data;
	}
}

/*

    def multiple(self, *args):
        if isinstance(args[0], str):
            data = []
            for i in args[1].decoded_line:
                data.append(i[1].execute(self.global_variable, self.global_class))
            return data[int(args[0])]
        return [i[1].execute(self.global_variable, self.global_class) for i in args[0].decoded_line]

    def while_pro(self, *arg):
        print("no")


    
		*/